src/traffic_server/InkAPI.cc

```diff
@@ -9004,6 +9005,169 @@ TSSslContextDestroy(TSSslContext ctx)
+{
+  if (nullptr == cert_path) {
+    return TS_ERROR;
+  }
+  key_path = key_path ? key_path : cert_path;
+
+  X509 *cert            = nullptr;
+  SSL_CTX *test_ctx     = nullptr;
+  SSL_CTX *target_ctx   = nullptr;
+  SSLCertContext *cc    = nullptr;
+  SSLCertLookup *lookup = SSLCertificateConfig::acquire();
+
+  if (nullptr != lookup) {
+    // Read cert from path to extract lookup key (common name)
+    scoped_BIO bio(BIO_new_file(cert_path, "r"));
+    if (bio) {
+      cert = PEM_read_bio_X509(bio.get(), nullptr, nullptr, nullptr);
+    }
+    if (!bio || !cert) {
+      SSLError("Failed to load certificate/key from %s", cert_path);
+      return TS_ERROR;
+    }
+
+    // Extract common name
+    int pos                       = X509_NAME_get_index_by_NID(X509_get_subject_name(cert), NID_commonName, -1);
+    X509_NAME_ENTRY *common_name  = X509_NAME_get_entry(X509_get_subject_name(cert), pos);
+    ASN1_STRING *common_name_asn1 = X509_NAME_ENTRY_get_data(common_name);
+    char *common_name_str         = reinterpret_cast<char *>(const_cast<unsigned char *>(ASN1_STRING_get0_data(common_name_asn1)));
+    if (ASN1_STRING_length(common_name_asn1) != static_cast<int>(strlen(common_name_str))) {
+      // Embedded NULL char
+      return TS_ERROR;
+    }
+    Debug("ssl.cert_update", "Updating from %s with common name %s", cert_path, common_name_str);
+
+    // Update context to use cert
+    cc = lookup->find(common_name_str);
+    if (cc && cc->ctx) {
+      // First try to update on a default context to verify
+      test_ctx = SSLDefaultServerContext();
+      if (!SSL_CTX_use_certificate(test_ctx, cert)) {
+        SSLError("Failed to assign cert from %s to SSL_CTX", cert_path);
+        X509_free(cert);
+        SSLReleaseContext(test_ctx);
+        return TS_ERROR;
+      }
+      if (!SSL_CTX_use_PrivateKey_file(test_ctx, key_path, SSL_FILETYPE_PEM)) {
+        SSLError("Failed to load server private key from %s", key_path);
+        SSLReleaseContext(test_ctx);
+        return TS_ERROR;
+      }
+      if (!SSL_CTX_check_private_key(test_ctx)) {
+        SSLError("server private key does not match the certificate public key for %s", cert_path);
+        SSLReleaseContext(test_ctx);
+        return TS_ERROR;
+      }
+      // update on the context found from lookup table
+      target_ctx = cc->ctx;
+      if (!SSL_CTX_use_certificate(target_ctx, cert)) {
+        SSLError("Failed to assign cert from %s to SSL_CTX in lookup table", cert_path);
+        return TS_ERROR;
+      }
+      if (!SSL_CTX_use_PrivateKey_file(target_ctx, key_path, SSL_FILETYPE_PEM)) {
+        SSLError("Failed to load server private key from %s to SSL_CTX in lookup table", key_path);
+        return TS_ERROR;
+      }
+      if (!SSL_CTX_check_private_key(target_ctx)) {
+        SSLError("Server private key does not match the certificate public key for %s", cert_path);
+      }
+      SSLReleaseContext(test_ctx);
+      return TS_SUCCESS;
```

Comment:
Are you actually adding/updating the server case?  Probably need to add some locking here too.